gprofiler/profilers/java.py

```diff
@@ -279,6 +280,72 @@ def get_ap_version() -> str:
     return Path(resource_path("java/async-profiler-version")).read_text()
 
 
+@functools.lru_cache(maxsize=1024)
+def get_jvm_flags(
+    process: Process,
+) -> List[Tuple[Optional[str], Optional[str], Optional[str], str, Optional[List[str]]]]:
+    return [
+        (flag_name, flag_type, flag_value, flag_source, flag_kind)
+        for flag_name, flag_type, flag_value, flag_source, flag_kind in get_jvm_flags_raw(process)
+        if filter_jvm_flags(flag_name, flag_type, flag_value, flag_source, flag_kind) is True
+    ]
+
+
+def filter_jvm_flags(
+    _flag_name: Optional[str],
+    flag_type: Optional[str],
+    _flag_value: Optional[str],
+    flag_source: str,
+    flag_kind: Optional[List[str]],
+) -> bool:
+    if flag_source == "default":
+        return False
+
+    if flag_type and flag_type not in ["bool", "int", "uint", "intx", "uintx", "uint64_t", "size_t", "double"]:
+        return False
+
+    if flag_kind and any(
+        (not_supported_kind in flag_kind) for not_supported_kind in ["notproduct", "pd", "experimental"]
+    ):
+        return False
+
+    return True
+
+
+def get_jvm_flags_raw(
+    process: Process,
+) -> List[Tuple[Optional[str], Optional[str], Optional[str], str, Optional[List[str]]]]:
+    try:
+        output = run_process([jattach_path(), str(process.pid), "jcmd", "VM.flags -all"]).stdout.decode()
+    except CalledProcessError as e:
+        logging.warning(f"Couldn't get jvm flags for process {process.pid}: {e.stderr}")
+        return []
+
+    vm_flags = []
+    vm_flags_pattern = re.compile(r"(\S+)\s+(\S+)\s+(:)?= (\S*)\s+{(.+?)}(?:\s*{(.*)})*")
+
+    for line in output.splitlines():
+        match = vm_flags_pattern.search(line)
+        if match:
+            flag_type = match.group(1)
+            flag_name = match.group(2)
+            flag_is_non_default_source_only_jdk_8 = match.group(3) == ":"
+            flag_value = match.group(4)
+            flag_kind = match.group(5).split()
+            flag_source_jdk_9 = match.group(6)
```

Comment:
Looks like [the terminology in OpenJDK is "flag origin"](https://github.com/openjdk/jdk/blob/544c16e0bdd4335b2624158fd1f6521984aa5079/src/hotspot/share/runtime/flags/jvmFlag.hpp#L36), let's rename accordingly.